PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py

```diff
@@ -0,0 +1,87 @@
+import ROOT
+
+from PhysicsTools.Heppy.analyzers.core.Analyzer import Analyzer
+from PhysicsTools.Heppy.analyzers.core.AutoHandle import AutoHandle
+from PhysicsTools.HeppyCore.utils.deltar import matchObjectCollection, matchObjectCollection3
+import PhysicsTools.HeppyCore.framework.config as cfg
+
+
+class TwoObjectsClass():
+    def __init__(self,pt=-1,phi=-10):
+        self.pt_ = pt
+        self.phi_ = phi
+    def pt(self):
+        return self.pt_
+    def phi(self):
+        return self.phi_
+"""
+L1TriggerAnalyzer is an Analyzer that .....
+"""
+class L1TriggerAnalyzer( Analyzer ):
+    def __init__(self, cfg_ana, cfg_comp, looperName ):
+        super(L1TriggerAnalyzer,self).__init__(cfg_ana,cfg_comp,looperName)
+        self.processName  = getattr(self.cfg_ana,"processName",("HLT"))
+        self.l1JetInputTag  = getattr(self.cfg_ana,"l1JetInputTag",("caloStage2Digis","Jet",self.processName))
+        self.l1TauInputTag  = getattr(self.cfg_ana,"l1JetInputTag",("caloStage2Digis","Tau",self.processName))
+        self.l1EGammaInputTag  = getattr(self.cfg_ana,"l1JetInputTag",("caloStage2Digis","EGamma",self.processName))
+        self.l1EtSumInputTag  = getattr(self.cfg_ana,"l1JetInputTag",("caloStage2Digis","EtSum",self.processName))
+        self.l1MuonInputTag  = getattr(self.cfg_ana,"l1JetInputTag",("gmtStage2Digis","Muon",self.processName))
+        
+    def declareHandles(self):
+        super(L1TriggerAnalyzer, self).declareHandles()
+        self.handles['l1tJets']  = AutoHandle( self.l1JetInputTag, 'BXVector<l1t::Jet>' )
+        self.handles['l1tTaus']  = AutoHandle( self.l1TauInputTag, 'BXVector<l1t::Tau>' )
+        self.handles['l1tEGammas']  = AutoHandle( self.l1EGammaInputTag, 'BXVector<l1t::EGamma>' )
+        self.handles['l1tEtSums']  = AutoHandle( self.l1EtSumInputTag, 'BXVector<l1t::EtSum>' )
+        self.handles['l1tMuons']  = AutoHandle( self.l1MuonInputTag, 'BXVector<l1t::Muon>' )
+
+    def beginLoop(self, setup):
+        super(L1TriggerAnalyzer,self).beginLoop(setup)
+
+    def process(self, event):
+        self.readCollections( event.input )
+        event.l1Jets = []
+        event.l1Taus = []
+        event.l1Muons = []
+        event.l1EGammas = []
+        event.l1MET2,event.l1MET,event.l1ET,event.l1MHT,event.l1HT = tuple([TwoObjectsClass()]*5)
+        try:
```

Comment:
if it fails at first event should it always fail? if so, can you add a flag in the except and avoid the "try" in later events?